backend/origuide/fold/fold_producer.py

```diff
@@ -27,16 +29,20 @@ def next_transition(self):
                 "vertices_coords": self.transitions[-1]["vertices_coords"]
             }
         )
+        self.last_steady_frame = len(self.transitions) - 1
+        self.encoder.next_step()
 
-    def accept(self, vertices: List[Vertex]):
-        self.transitions.append(
-            {
-                "frame_inherit": True,
-                "frame_parent": len(self.transitions) - 1,
-                "vertices_coords": list(map(lambda x: list(x.pos.vec), vertices))
-            }
-        )
+    def accept(self, vertices: List[Vertex], last_step=False):
+        frame = {
+            "frame_inherit": True,
+            "frame_parent": len(self.transitions) - 1,
+            "vertices_coords": list(map(lambda x: list(x.pos.vec), vertices))
+        }
+        encoded = self.encoder.encode(frame, last_step)
+        self.encoder.advance()
```

Comment:
Do we ever encode multiple times to advance? Maybe the logic could be hidden in .encode()?

Reply:
I wanted to split these 2 functions in interface, so as to allow skipping frames by calling advance, without calling encode. Could potentially prove useful, maybe.